lib/wasabi/parser.rb

```diff
@@ -260,8 +260,13 @@ def input_output_for(operation, input_output)
         end
 
         message_ns_id, message_type = nil
-        message_ns_id = port_message_ns_id
-        message_type = port_message_type
+
+        soap_operation = operation.element_children.find { |node| node.name == 'operation' }
+
+        if soap_operation.nil? || soap_operation['style'] != 'rpc'
```

Comment:
mind explaining the change here? why do we branch into here only if nil soap operation or the style is (presumably) document

Reply:
This could be refactored to `unless soap_operation&.get_attribute('style') == 'rpc'`

It just disables the behavior introduced in 4.0 for RPC operations. When the value is `nil` or `document` it keeps the current behavior.